File: C Programs/Problems on Pattern Printing/Pattern1.c
/* Write a program which accept number from user and display below pattern.

Input : 5
Output : * * * * * # # # # #

Input : 6
Output : * * * * * * # # # # # # #

Input : -5
Output : * * * * * # # # # #

Input : 2
Output : * * # #           

*/

#include<stdio.h>

//////////////////////////////////////////////////////////////
//
//  Function name :  Display
//  Input : Interger
//  Output :  None
//  Description :   It is used to Display display pattern
//  Date :  9th August 2020
//
//////////////////////////////////////////////////////////////

void Display(int iNo)
{
    int iCnt=0;

    if(iNo==0)
    {
        printf("Invalid Input");
        return;
    }
    if(iNo<0)
    {
        iNo=-iNo;
    }

    for(iCnt=1;iCnt<=iNo;iCnt++)
        {
                printf(" * ");
        }

    for(iCnt=1;iCnt<=iNo;iCnt++)
        {
                printf(" # ");
        }
}

int main()
{
    int  iValue = 0;

    printf("Enter starting point: ");
    scanf("%d",&iValue);

    Display(iValue);
    return 0;
}

File: Java Programs/Problems on Pattern Printing/Alternate # and * pattern/Pattern 1.java
/* Write Java program which accept number of rows and number of columns from user and display below pattern.

Input : iRow = 3 iCol = 4
Output : 
 * # * #
 * # * #
 * # * # 

*/

import java.lang.*;
import java.util.*;

class Pattern
{
    void PrintPattern(int iRow,int iCol)
     {
       int i=0;
       int j=0;

       for(i=1;i<=iRow;i++)
        {
           for(j=1;j<=iCol;j++)
            {
               if(j%2==0)
                {
                   System.out.print("#"+"\t");
                }
               else
                {
                   System.out.print("*"+"\t");
                }
           }
               System.out.println();
       }
   }
    
}

class Demo
{
    public static void main(String arg[])
     {
        Scanner sobj=new Scanner(System.in);
        Pattern pobj=new Pattern();
        
        System.out.println("Enter no of rows: ");
        int iRow=sobj.nextInt();
        
        System.out.println("Enter no of Columns: ");
        int iCol=sobj.nextInt();
        
        pobj.PrintPattern(iRow,iCol);
    }
}
